src/geometry.rs

```diff
@@ -174,9 +177,25 @@ impl Geometry {
         &self.normals
     }
 
-    pub fn edges_iter<'a>(&'a self) -> impl Iterator<Item = HalfEdge> + 'a {
+    pub fn oriented_edges_iter<'a>(&'a self) -> impl Iterator<Item = OrientedEdge> + 'a {
+        self.triangle_faces_iter()
+            .flat_map(|face| ArrayVec::from(face.to_oriented_edges()).into_iter())
+    }
+
+    pub fn unoriented_edges_iter<'a>(&'a self) -> impl Iterator<Item = UnorientedEdge> + 'a {
         self.triangle_faces_iter()
-            .flat_map(|face| ArrayVec::from(face.to_edges()).into_iter())
+            .flat_map(|face| ArrayVec::from(face.to_unoriented_edges()).into_iter())
+    }
+
+    /// Genus of a mesh is the number of holes in topology / conectivity
+    /// The mesh must be triangular and watertight
+    /// V - E + F = 2 (1 - G)
+    pub fn mesh_genus(&self, edges: &HashSet<UnorientedEdge>) -> i32 {
```

Comment:
This shouldn't be a method on `Geometry` if you need to provide external data, possibly coming from another mesh. Let's make it a function in `mesh_analysis.rs`. Also the fact it takes a hash set only to get it's `len` is worrisome.

Reply:
>Also the fact it takes a hash set only to get it's `len` is worrisome.

what do you suggest? asking for the length only would move the logic outside of the function. also the same applies to `&self` - also needed just the lengths

Reply:
> This shouldn't be a method on Geometry if you need to provide external data, possibly coming from another mesh.

OK

Reply:
> what do you suggest?

My bad: we can just compute the number of edges each time as we already have the data. Let's keep this a method and just take the count from the iterator you can readily spawn.